test/xunit.analyzers.tests/TheoryMethodMustUseAllParametersTests.cs

```diff
@@ -0,0 +1,140 @@
+
+            for (int i = 0; i < messageArgs.Length; i++)
+            {
+                var (method, type, parameter) = messageArgs[i];
+                string message = $"Theory method '{method}' on test class '{type}' does not use parameter '{parameter}'.";
+
+                var diagnostic = diagnosticArray[i];
+                Assert.Equal(message, diagnostic.GetMessage());
+                Assert.Equal("xUnit1026", diagnostic.Id);
+                Assert.Equal(DiagnosticSeverity.Warning, diagnostic.Severity);
+            }
+        }
+
+        [Fact]
+        public async void FindsError_ParameterNotReferenced()
+        {
+            var diagnostics = await CodeAnalyzerHelper.GetDiagnosticsAsync(Analyzer, @"
+using Xunit;
+
+class TestClass
+{
+    [Theory]
+    void TestMethod(int unused) { }
+}");
+
+            CheckDiagnostics(diagnostics,
+                (method: "TestMethod", type: "TestClass", parameter: "unused"));
+        }
+
+        [Fact]
+        public async void FindsError_ParameterUnread()
+        {
+            var diagnostics = await CodeAnalyzerHelper.GetDiagnosticsAsync(Analyzer, @"
+using System;
+using Xunit;
+
+class TestClass
+{
+    [Theory]
+    void TestMethod(int unused)
+    {
+        unused = 3;
+    }
+}");
+
+            CheckDiagnostics(diagnostics,
+                (method: "TestMethod", type: "TestClass", parameter: "unused"));
+        }
+
+        [Fact]
+        public async void FindsError_MultipleUnreadParameters()
+        {
+            var diagnostics = await CodeAnalyzerHelper.GetDiagnosticsAsync(Analyzer, @"
+using Xunit;
+
+class TestClass
+{
+    [Theory]
+    void TestMethod(int foo, int bar, int baz) { }
+}");
+
+            CheckDiagnostics(diagnostics,
+                (method: "TestMethod", type: "TestClass", parameter: "foo"),
+                (method: "TestMethod", type: "TestClass", parameter: "bar"),
+                (method: "TestMethod", type: "TestClass", parameter: "baz"));
+        }
+
+        [Fact]
+        public async void FindsError_SomeUnreadParameters()
+        {
+            var diagnostics = await CodeAnalyzerHelper.GetDiagnosticsAsync(Analyzer, @"
+using System;
+using Xunit;
+
+class TestClass
+{
+    [Theory]
+    void TestMethod(int foo, int bar, int baz)
+    {
+        Console.WriteLine(bar);
+        baz = 3;
+    }
+}");
+
+            CheckDiagnostics(diagnostics,
+                (method: "TestMethod", type: "TestClass", parameter: "foo"),
+                (method: "TestMethod", type: "TestClass", parameter: "baz"));
+        }
+
+        [Fact]
+        public async void DoesNotFindError_ParameterRead()
+        {
+            var diagnostics = await CodeAnalyzerHelper.GetDiagnosticsAsync(Analyzer, @"
+using System;
+using Xunit;
+
+class TestClass
+{
+    [Theory]
+    void TestMethod(int unused)
+    {
+        Console.WriteLine(unused);
+    }
+}");
+        }
+
+        [Fact]
+        public async void DoesNotFindError_ParameterAddressTaken()

```

Comment:
I don't need this test actually, I only added it because I assumed `DFA.ReadInside` wouldn't include `unused` here and I would have to union it with `DFA.UnsafeAddressTaken` in the analyzer. Turns out I was wrong.